examples/threading/threading.c

```diff
@@ -8,13 +8,32 @@
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
+#define MILLI_TO_MICRO 1000
+
 void* threadfunc(void* thread_param)
 {
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    
+    (thread_func_args->thread_complete_success) =false;
+    
+    usleep((thread_func_args->wait_to_obtain_ms) *MILLI_TO_MICRO);
+    
+    pthread_mutex_lock(thread_func_args->mutex);
+    
+    
+   //Makinf thread to sleep uninterruptedly by locking it
+    usleep((thread_func_args->wait_to_release_ms) *MILLI_TO_MICRO);
+    
+    pthread_mutex_unlock(thread_func_args->mutex);
+    
+    //Successfully updating the flag as the thread completed
+   (thread_func_args->thread_complete_success) =true;  
```

Comment:
@jasan2303 what will happen in case of mutex lock/unlock failure or usleep failure?